zookeeper-server/src/test/java/org/apache/zookeeper/common/X509UtilTest.java

```diff
@@ -406,26 +407,30 @@ public void testGetSslHandshakeDetectionTimeoutMillisProperty() {
         }
     }
 
+    @Test(expected = X509Exception.SSLContextException.class)
+    public void testCreateSSLContext_invalidCustomSSLContextClass() throws Exception {
+        ZKConfig zkConfig = new ZKConfig();
+        ClientX509Util clientX509Util = new ClientX509Util();
+        zkConfig.setProperty(clientX509Util.getSslContextSupplierClassProperty(), String.class.getCanonicalName());
+        clientX509Util.createSSLContext(zkConfig);
+    }
+
     @Test
-    public void testCreateSSLContext_invalidCustomSSLContextClass() {
+    public void testCreateSSLContext_validNullCustomSSLContextClass() throws X509Exception.SSLContextException {
         ZKConfig zkConfig = new ZKConfig();
         ClientX509Util clientX509Util = new ClientX509Util();
-        zkConfig.setProperty(clientX509Util.getSslClientContextProperty(), String.class.getCanonicalName());
-        try {
-            clientX509Util.createSSLContext(zkConfig);
-            fail("SSLContextException expected.");
-        } catch (X509Exception.SSLContextException e) {
-            assertTrue(e.getMessage().contains(clientX509Util.getSslClientContextProperty()));
-        }
+        zkConfig.setProperty(clientX509Util.getSslContextSupplierClassProperty(), X509UtilTest.class.getCanonicalName() + "$" + NullSslContextSupplier.class.getSimpleName());
+        final SSLContext sslContext = clientX509Util.createSSLContext(zkConfig);
+        assertNull(sslContext);
     }
 
     @Test
     public void testCreateSSLContext_validCustomSSLContextClass() throws X509Exception.SSLContextException {
         ZKConfig zkConfig = new ZKConfig();
         ClientX509Util clientX509Util = new ClientX509Util();
-        zkConfig.setProperty(clientX509Util.getSslClientContextProperty(), ZKTestClientSSLContext.class.getCanonicalName());
+        zkConfig.setProperty(clientX509Util.getSslContextSupplierClassProperty(), X509UtilTest.class.getCanonicalName() + "$" + SslContextSupplier.class.getSimpleName());
         final SSLContext sslContext = clientX509Util.createSSLContext(zkConfig);
-        assertNull(sslContext);
+        assertNotNull(sslContext);
```

Comment:
nit: I think this can be `assertEquals(SSLContext.getDefault(), sslContext)`

Reply:
Good point - getDefault() returns a singleton representing the default SSLContext, so it'll be the same object.